tools/generator/src/PBXTarget+Extensions.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+import XcodeProj
+
+public extension PBXTarget {
+    var buildableName: String {
+        return productName ?? name
```

Comment:
All of the `PBXTarget` instances that I have seen have had a value for `productName`. In terms of whether it should default to name, I think that it is reasonable given the following:
- We need a value for `buildableName`
- The `productName` is optional
- The `productName` seems to best align with the values that are generated by Xcode itself
- The `name` value still conveys the intent given that we opted for one scheme per target

Reply:
I mean, I think `buildableName` needs to reference a `productName`. I think we can't use `name` because the scheme won't actually work as expected then.

I would rather we return `nil` here and then throw where we try to use the nil value.

Reply:
To avoid duplicating the check logic, I replaced `buildableName` with `getBuildableName`. This resulted in `schemeName` changing to `getSchemeName`.